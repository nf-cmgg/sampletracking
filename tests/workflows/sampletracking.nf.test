nextflow_workflow {

    name "Test Workflow SAMPLETRACKING"
    script "workflows/sampletracking.nf"

    tag "workflows"
    tag "workflows/sampletracking"

    workflow "SAMPLETRACKING"

    test("Should run without failures") {
        options "-stub"
        setup {
            run("BWA_INDEX") {
                script "../../modules/nf-core/bwa/index/main.nf"
                process {
                    """
                    input[0] = [
                        [:],
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna", checkIfExists: true)
                    ]
                    """

                }
            }
        }
        when {
            params {
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [id:"sample1", pool: "pool1", pool_count:2], // meta
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/exome/bam/sample1.sort.bam", checkIfExists:true),     // sample_bam
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/exome/bam/sample1.sort.bam.bai", checkIfExists:true), // sample_index
                        [
                            file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/snp/fastq/snp_sample1_R1.fastq.gz"),
                            file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/snp/fastq/snp_sample1_R2.fastq.gz"),
                        ],      // snp_fastq
                        [],     // snp_bam
                        []      // snp_bai
                    ],
                    [
                        [id:"sample2", pool: "pool1", pool_count:2], // meta
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/exome/cram/sample2.sorted.cram", checkIfExists:true),        // sample_bam
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/exome/cram/sample2.sorted.cram.crai", checkIfExists:true),   // sample_index
                        [],      // snp_fastq
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/snp/cram/snp_sample2.sorted.cram"),           // snp_bam
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/snp/cram/snp_sample2.sorted.cram.crai"),     // snp_bam
                    ],
                )
                input[1] = BWA_INDEX.out.index
                input[2] = Channel.value(
                    [
                        [:],
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna"),
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai")
                    ]
                )
                input[3] = Channel.value(
                    [
                        [:],
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/genome/picard/haplotype_map.txt")
                    ]
                )
                input[4] = params.outdir
                input[5] = ""
                input[6] = ""
                input[7] = ""
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("Should run without failures - skip crosscheck fingerprints") {
        options "-stub"

        when {
            params {
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [id:"sample1", pool: "pool1", pool_count:2], // meta
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/exome/bam/sample1.sort.bam", checkIfExists:true),     // sample_bam
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/exome/bam/sample1.sort.bam.bai", checkIfExists:true), // sample_index
                        [],     // snp_fastq
                        [],     // snp_bam
                        []      // snp_bai
                    ],
                    [
                        [id:"sample2", pool: "pool1", pool_count:2], // meta
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/exome/cram/sample2.sorted.cram", checkIfExists:true),        // sample_bam
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/exome/cram/sample2.sorted.cram.crai", checkIfExists:true),   // sample_index
                        [],      // snp_fastq
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/snp/cram/snp_sample2.sorted.cram"),           // snp_bam
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/illumina/snp/cram/snp_sample2.sorted.cram.crai"),     // snp_bam
                    ],
                )
                input[1] = []
                input[2] = Channel.value(
                    [
                        [:],
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna"),
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai")
                    ]
                )
                input[3] = Channel.value(
                    [
                        [:],
                        file("https://github.com/nf-cmgg/test-datasets/raw/sampletracking/data/genomics/homo_sapiens/genome/picard/haplotype_map.txt")
                    ]
                )
                input[4] = params.outdir
                input[5] = ""
                input[6] = ""
                input[7] = ""
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }
}
