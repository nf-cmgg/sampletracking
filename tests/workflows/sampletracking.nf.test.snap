{
    "Should run without failures - skip crosscheck fingerprints": {
        "content": [
            {
                "0": [
                    "multiqc_report.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "pool1_multiqc_report.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    
                ],
                "3": [
                    [
                        {
                            "id": "sample1",
                            "pool": "pool1",
                            "pool_count": 2
                        },
                        "sample1.xy_xy.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample1.xy_sry.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample1.xy_hetx.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "sample2",
                            "pool": "pool1",
                            "pool_count": 2
                        },
                        "sample2.xy_xy.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample2.xy_sry.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample2.xy_hetx.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,dac5a146f64564be62294d29fe21cb67"
                ],
                "crosscheck_metrics": [
                    
                ],
                "multiqc_pools": [
                    "pool1_multiqc_report.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "multiqc_report": [
                    "multiqc_report.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "sex_prediction": [
                    [
                        {
                            "id": "sample1",
                            "pool": "pool1",
                            "pool_count": 2
                        },
                        "sample1.xy_xy.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample1.xy_sry.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample1.xy_hetx.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "sample2",
                            "pool": "pool1",
                            "pool_count": 2
                        },
                        "sample2.xy_xy.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample2.xy_sry.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample2.xy_hetx.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dac5a146f64564be62294d29fe21cb67"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-21T11:08:41.920631027"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "multiqc_report.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "pool1_multiqc_report.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    [
                        {
                            "groupSize": 2,
                            "groupTarget": {
                                "id": "pool1"
                            }
                        },
                        "pool1.crosscheck_metrics.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "sample1",
                            "pool": "pool1",
                            "pool_count": 2
                        },
                        "sample1.xy_xy.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample1.xy_sry.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample1.xy_hetx.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "sample2",
                            "pool": "pool1",
                            "pool_count": 2
                        },
                        "sample2.xy_xy.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample2.xy_sry.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample2.xy_hetx.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,7442362a1e457dad7ce796c703bd6380",
                    "versions.yml:md5,ab16dda2f91f60705355cab06b0145bf",
                    "versions.yml:md5,dac5a146f64564be62294d29fe21cb67"
                ],
                "crosscheck_metrics": [
                    [
                        {
                            "groupSize": 2,
                            "groupTarget": {
                                "id": "pool1"
                            }
                        },
                        "pool1.crosscheck_metrics.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "multiqc_pools": [
                    "pool1_multiqc_report.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "multiqc_report": [
                    "multiqc_report.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "sex_prediction": [
                    [
                        {
                            "id": "sample1",
                            "pool": "pool1",
                            "pool_count": 2
                        },
                        "sample1.xy_xy.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample1.xy_sry.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample1.xy_hetx.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "sample2",
                            "pool": "pool1",
                            "pool_count": 2
                        },
                        "sample2.xy_xy.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample2.xy_sry.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "sample2.xy_hetx.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7442362a1e457dad7ce796c703bd6380",
                    "versions.yml:md5,ab16dda2f91f60705355cab06b0145bf",
                    "versions.yml:md5,dac5a146f64564be62294d29fe21cb67"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-21T11:08:18.733645178"
    }
}